domains/auth/Auth.API/Tests/Controllers/TestAuthController.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.Linq;
+using System.Text.Json;
+using API.Configuration;
+using API.Models;
+using API.Services;
+using API.Services.OidcProviders;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Moq;
+using Xunit;
+using Xunit.Categories;
+
+namespace API.Controllers;
+
+[UnitTest]
+public class TestAuthController
+{
+    private readonly Mock<IOidcProviders> _mockSignaturGruppen = new();
+    private readonly Mock<ILogger<AuthController>> logger = new();
+    private readonly Mock<ITokenStorage> tokenStorage = new();
+    private readonly Mock<IOptions<AuthOptions>> authOptionsMock = new();
+    private readonly Mock<ICryptographyService> _cryptographyService = new();
+    private readonly InvalidateAuthStateValidator _validator = new();
+
+    private AuthController _authController;
+
+    public TestAuthController()
+    {
+        authOptionsMock.Setup(x => x.Value).Returns(new AuthOptions
+        {
+            CookieName = "Authorization",
+        });
+
+        _authController = new AuthController(logger.Object, _mockSignaturGruppen.Object, authOptionsMock.Object,
+            tokenStorage.Object, _cryptographyService.Object, _validator)
+        {
+            ControllerContext = new ControllerContext()
+            {
+                HttpContext = new DefaultHttpContext()
+            }
+        };
+    }
+
+    [Fact]
+    public void CanInvalidateToken()
+    {
+        var authState = new AuthState()
+        {
+            IdToken = "test"
+        };
+
+        var authStateAsString = JsonSerializer.Serialize(authState);
+
+        _cryptographyService
+            .Setup(x => x.Decrypt(authStateAsString))
+            .Returns(authStateAsString);
+
+        var response = _authController.Invalidate(authStateAsString);
+
+        _mockSignaturGruppen.Verify(mock => mock.Logout(authState.IdToken), Times.Once);
+        Assert.IsType<OkResult>(response);
+    }
+
+    [Fact]
+    public void ReturnBadRequestWhenNoState()
+    {
+        var response = _authController.Invalidate("");
+
+        Assert.IsType<BadRequestObjectResult>(response);
+        var result = response as BadRequestObjectResult;
+        Assert.Equal("Cannot decrypt " + nameof(AuthState), result?.Value);
+    }
+
+    [Fact]
+    public void ReturnBadRequestWhenNoToken()
+    {
+        var authState = new AuthState();
+        var authStateAsString = JsonSerializer.Serialize(authState);
+        _cryptographyService
+            .Setup(x => x.Decrypt(authStateAsString))
+            .Returns(authStateAsString);
+
+        var response = _authController.Invalidate(authStateAsString);
+
+        Assert.IsType<BadRequestObjectResult>(response);
+        var result = response as BadRequestObjectResult;
+        Assert.Equal(nameof(AuthState.IdToken) + " must not be null", result?.Value);
```

Comment:
I would argument against having tests fail when we change an exception message. (I also have a suggested change for this message :P )